Quality_Declaration_fastrtps.md

```diff
@@ -0,0 +1,137 @@
+# **External Dependency Quality declaration** `fastrtps` 
+
+This document is a declaration of software quality for the `fastrtps` external dependency, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/rep-2004/rep-2004.rst).
+
+As stated in their public repository, eprosima Fast RTPS is a C++ implementation of the RTPS (Real Time Publish Subscribe) protocol, which provides publisher-subscriber communications over unreliable transports such as UDP, as defined and maintained by the Object Management Group (OMG) consortium.
+
+## Summary
+
+The `fastrtps` as a complete software solution manages to meet most of the ROS2 package requirements for Quality Level 1, this includes, but not limited to, a complete established workflow for new changes in its repository, testing for the features of the software and documentation with examples.
+
+As for the missing topics, eProsima does not state their versioning policies for the repository and does not state explicitly the API/ABI policies for the code in `fastrtps`. As the code deals with the standard RPTS implementation, it is not expected that the basic API functionalities will be changed. In any case, the code is actively being tested against their [CI](http://jenkins.eprosima.com:8080/) and in the [ROS2 CI](https://ci.ros2.org/).
+
+There is no coverage information summary/report and, apparently, no checks being made in terms of linters/static analysis, however, as the core parts used in ROS2 are actively being tested under the packages that use `fastrtps` (`rmw_fastrtps_cpp, rmw_fastrtps_shared_cpp, rosidl_typesupport_fastrtps_c, rosidl_typesupport_fastrtps_cpp`).
+
+eProsima states publicly that their RTPS implementation is used as the default middleware for ROS2. This shows how important is for them to keep their software being high quality and compatible with ROS2 packages.
+  
+Considering the previously mentioned reasons, we consider this library to be robust and reliable, and hence we declare it to qualify as a level 1 external dependency.
+
+
+## Comparison with ROS packages quality standards
+
+### Version policy
+
+ 1. *Must have a version policy*
```

Comment:
Use the Quality declaration template https://github.com/ros-infrastructure/rep/blob/4a3bbf6c296a725fda742ff8e695e2e1f6f59aa9/rep-2004.rst#quality-declaration-template